Timeline: zoom/scaling

MattiasBeming opened the pull request:
In tests; now ignores most common files such ass css-files and others, for example jpg, png

Closes #22

MattiasBeming committed:
feat: implement timeline navigation and zoom

Component placed containing scrollable timeline and dropdown menu. Dropdown menu is where scaling options exist to decide how many hours should be visible. Lines drawn to indicate hours.

MattiasBeming committed:
test: add unit-test for function zoom in timeline

Also tests getLinePlacements in timeline

MattiasBeming committed:
fix: move mock-files to frontend folder

__mocks__/ were located in root of project, it should be in frontend/

MattiasBeming commented:
A miscommunication led to a problem (pull-request #163 got accepted to early). Which in turn led to more confusion. The latest commit (a726fed) fixed the problem.

This is why e4f8a91 fails the test.

sporreking reviewed (changes requested):
Good job! A few changes:

**timeline.module.css**
* remove left, right, height, bottom properties from `.main`
* add `width: 80vw` and `height: 35vh` to `.main`
* change position to relative in `.topbar`, `.dropdown` and `.sliderbox`
* add `top: 0` to `.line`

**timeline.js**
* add space at beginning of line 12-16
* add `@param` tag to getLinePlacements doc-comment
* fix spacing in `getLinePlacements` code
* comment HTML (use `{/* comment */}`)
* break line in style attribute on line 54
* put semicolon on line 72-73, and 80-81
* remove `funcDummy` from dispatch->prop mapping
* remove logScaling import

**stateTimeline.js**
* remove LOG_SCALING action constant, creator, and handling in reducer
* add semicolon to line 16, 22, 29, 36, 43, 51, and 69
* comment operation in ZOOM case (reducer)
* row 15: change comment to 36 hours

**timeline.test.js**
* semicolons at row 122, 150, 1353

MattiasBeming committed:
fix: feedback

Changed css for timeline. Added comments for most things in timeline. Syntax fixes. Removed log_scaling function.

sporreking reviewed (approved):
Everything looks good! Good job!
